client/state/plugins/wporg/reducer.js

```diff
@@ -16,6 +26,27 @@ function updatePluginState( state = {}, pluginSlug, attributes ) {
 	);
 }
 
+const defaultPluginsListState = ( {
+	shortLists: {},
+	fullLists: {},
+	currentSearchTerm: null,
+	fetching: {},
+	lastFetchedPage: {}
+} );
+
+function updatePluginsList( state = defaultPluginsListState, category, page, list ) {
+	if ( ! page || page === _DEFAULT_FIRST_PAGE ) {
+		state.shortLists[ category ] = Object.assign( [], list.slice( 0, 6 ) );
```

Comment:
notice that by directly mutating `state` here we will be making a change without changing the `value` of `state`, thus all pure components which only update on state changes won't update as a result of this change.

instead, we need to be splitting reducers into different units and composing them together. I'd be more than happy to give a short demo/workshop